cobra/flux_analysis/find_active_reactions.py

```diff
@@ -0,0 +1,602 @@
+
+"""
+Find all active reactions by solving a single MILP problem
+or a (small) number of LP problems
+
+"""
+import logging
+from cobra.flux_analysis.variability import flux_variability_analysis
+from cobra.flux_analysis.helpers import normalize_cutoff
+from optlang import Model, Variable, Constraint, Objective
+from optlang.symbolics import Zero
+from scipy.linalg import orth
+import numpy as np
+
+LOGGER = logging.getLogger(__name__)
+
+
+def relax_model_bounds(model, bigM=1e4):
+    """
+    Relax all upper and lower bounds in the model.
+    All positive upper bounds will become bigM.
+    All negative lower bounds will become -bigM.
+    All positive lower bounds and negative upper bounds will become zero.
+
+    Parameters
+    ----------
+    model: cobra.Model
+        cobra model. It will *not* be modified.
+    bigM: float, optional
+        a large constant for relaxing the model bounds, default 1e4.
+
+    Returns
+    -------
+    cobra.Model
+        cobra model with relaxed bounds
+
+    """
+
+    for r in model.reactions:
+        r.upper_bound = bigM if r.upper_bound > 0 else 0
+        r.lower_bound = -bigM if r.lower_bound < 0 else 0
+
+
+def find_active_reactions(model, bigM=10000, zero_cutoff=None,
+                          relax_bounds=True, solve="lp"):
+    """
+    Find all active reactions by solving a single MILP problem
+    or a (small) number of LP problems
+
+    Parameters
+    ----------
+    model: cobra.Model
+        cobra model. It will *not* be modified.
+    bigM: float, optional
+        a large constant for bounding the optimization problem, default 1e4.
+    zero_cutoff: float, optional
+        The cutoff to consider for zero flux
+        (default 1e-5 if relax_bounds is True, else model.tolerance).
+    relax_bounds: True or False
+        Whether to relax the model bounds.
+        All +ve LB set as 0, all -ve UB set as 0, all -ve LB set as -bigM,
+        all +ve UB set as bigM. If False, use the original bounds in the model.
+        Default True
+    solve: "lp", "milp" or "fastSNP"
+        - "lp":      to solve a number of LP problems
+                     (usually finish by solving 5 LPs)
+        - "milp":    to solve a single MILP problem
+        - "fastSNP": find a minimal nullspace basis using Fast-SNP and then
+                     return the reactions with nonzero entries in the nullspace
+        Default "lp". For models with numerical difficulties when using "lp"
+        or "milp", it is recommanded to tighten all tolerances:
+        feasbility, optimality and integrality
+
+    Returns
+    -------
+    list
+        List of reaction IDs which can carry flux.
+
+    Notes
+    -----
+    The optmization problem solved is as follow:
```

Comment:
It'd be really neat to use the Sphinx directives so that this is rendered as proper math. See [here](https://stackoverflow.com/a/11226398).